replace_using_package_version/replace_using_package_version.py

```diff
@@ -160,26 +161,45 @@ def apply_regex_to_file(input_file, output_file, regex, replacement):
 
 
 def find_package_version(package, rpm_dir):
+    version = parse_version('')
     try:
         version = get_pkg_version(package)
     except Exception:
-        # If package is not found in rpmdb check local repositories
-        version = parse_version('')
-        for root, _, files in os.walk(rpm_dir):
-            packages = [
-                f for f in files if f.endswith('rpm') and package in f
-            ]
-            for pkg in packages:
-                rpm_file = os.path.realpath(os.path.join(root, pkg))
-                if get_pkg_name_from_rpm(rpm_file) == package:
-                    rpm_version = get_pkg_version_from_rpm(rpm_file)
-                    if rpm_version >= version:
-                        version = rpm_version
+        version = find_package_version_in_local_repos(rpm_dir, package)
+
+    if not str(version):
+        version = find_package_version_in_obsinfo('.', package)
+
     if not str(version):
         raise Exception('Package version not found')
     return str(version)
 
 
+def find_package_version_in_local_repos(repo_path, package):
+    version = parse_version('')
+    for root, _, files in os.walk(repo_path):
+        packages = [
+            f for f in files if f.endswith('rpm') and package in f
+        ]
+        for pkg in packages:
+            rpm_file = os.path.join(root, pkg)
+            if get_pkg_name_from_rpm(rpm_file) == package:
+                rpm_ver = get_pkg_version_from_rpm(rpm_file)
+                if rpm_ver >= version:
+                    version = rpm_ver
+    return version
+
+
+def find_package_version_in_obsinfo(path, package):
+    version = parse_version('')
+    for f in os.listdir(path):
```

Comment:
Why do you use `os.listdir` here and `os.walk` in `find_package_version_in_local_repos`?

Reply:
This is because the `*.obsinfo` is at the top level of sources and this is applied on the current directory, I don't want to parse all the files in there as there could be tons of files. On the other side when parsing the repositories we needed recursive search as each repository is in a separate subfolder and we should not expect a repository tree including thousands or any other insane number of files.